sui_programmability/adapter/src/adapter.rs

```diff
@@ -405,7 +409,33 @@ pub fn store_package_and_init_modules<
 
     // wrap the modules in an object, write it to the store
     // The call to unwrap() will go away once we remove address owner from Immutable objects.
-    let package_object = Object::new_package(modules, ctx.digest());
+
+    // Must populate commitments to deps
+    let mut deps = BTreeSet::new();
+    for m in &modules {
+        let mut g: BTreeSet<ObjectID> = m
+            .module_handles
+            .iter()
+            .filter_map(|j| {
+                let r = ObjectID::from(*m.address_identifier_at(j.address));
+                if r == package_id {
+                    None
+                } else {
+                    Some(r)
+                }
+            })
+            .collect();
+        deps.append(&mut g);
+    }
+
+    // Get the hashes of all the external deps
+    // Okay to unwrap since dep objects must exist on chain for new package to exist
+    let deps: Vec<PackageObjectRef> = deps
+        .iter()
+        .map(|id| (*id, state_view.read_object(id).unwrap().digest()))
```

Comment:
I feel like calling `digest()` here could be costly, so we should visit the topic of caching the digest of immutable objects in the object itself..
The downside is an increased object size.

Reply:
If you do not trust the external source of objects then you must call `.digest()` since you cannot trust that a set digest field would be correct, right? Its only if you read the package from a trusted source (say your own store) that you can trust a stored digest field.